<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="joystick_linear_axis"   default="1" />
    <arg name="joystick_angular_axis"   default="0" />
    <arg name="joystick_camera_pan_axis"   default="2" />
    <arg name="joystick_camera_tilt_axis"   default="3" />
    <arg name="twist_publish_topic_name"   default="cmd_vel" />
    <arg name="deadzone_val"   default="0.05" />

    <node name="joystick" pkg="joy" type="joy_node" output="log" required="true">
        <param name="dev" value="/dev/input/js0" type="str"/>
        <param name="deadzone" value="$(arg deadzone_val)" />
        <param name="autorepeat_rate" value="30.0" />
    </node>
    <!-- <node name="joy_throttle" pkg="topic_tools" type="throttle" output="log" required="true" args="messages /joy 30 /joy_throttled" /> -->


    <group ns="rover6" >
        <node name="rover6_teleop" pkg="rover6_teleop" type="rover6_teleop_node.py" output="screen" required="true">
            <param name="deadzone_joy_val" value="$(arg deadzone_val)" />
            <param name="joystick_topic" value="/joy" />
            <param name="linear_scale" value="0.8" />
            <param name="angular_scale" value="2.5" />

            <param name="linear_axis" value="$(arg joystick_linear_axis)" />
            <param name="angular_axis" value="$(arg joystick_angular_axis)" />
            <param name="camera_pan_axis" value="$(arg joystick_camera_pan_axis)" />
            <param name="camera_tilt_axis" value="$(arg joystick_camera_tilt_axis)" />
        </node>
    </group>
</launch>
